crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -100,22 +101,18 @@ impl Diagnostic {
         Arc::make_mut(&mut self.inner).subs.push(sub);
     }
 
-    /// Print this diagnostic to the given writer.
+    /// Return a `std::fmt::Display` implementation that renders this
+    /// diagnostic into a human readable format.
     ///
-    /// # Errors
-    ///
-    /// This is guaranteed to only return an error if the underlying
-    /// writer given returns an error. Otherwise, the formatting of
-    /// diagnostics themselves is infallible.
-    pub fn print(
-        &self,
-        db: &dyn Db,
-        config: &DisplayDiagnosticConfig,
-        mut wtr: impl std::io::Write,
-    ) -> std::io::Result<()> {
+    /// Note that this `Display` impl includes a trailing line terminator, so
+    /// callers should prefer using this with `write!` instead of `writeln!`.
+    pub fn display<'c, 'r, 'd>(
+        &'d self,
+        db: &'r dyn Db,
```

Comment:
We almost exclusively use `'db` for the db lifetime. 
```suggestion
        db: &'db dyn Db,
```

Reply:
Yeah... I suppose this will become generic at some point, but it seems fine to just use `db` for now.